spec/ddtrace/tasks/exec_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'spec_helper'
+require 'ddtrace/tasks/exec'
+
+RSpec.describe Datadog::Tasks::Exec do
+  subject(:task) { described_class.new(args) }
+  let(:args) { ['ruby', '-e', '"RUBY_VERSION"'] }
+
+  describe '::new' do
+    it { is_expected.to have_attributes(args: args) }
+  end
+
+  describe '#run' do
+    subject(:run) { task.run }
+    let(:result) { double('result') }
+
+    around do |example|
+      # Make sure RUBYOPT is returned to its original state.
+      original_opts = ENV['RUBYOPT']
+      example.run
+      ENV['RUBYOPT'] = original_opts
```

Comment:
We likely need this `ENV` restoration to original to be inside an ensure block, as we don't know if `example.run` might raise an error and leave the environment in a bad state.

Reply:
Hmmm, fair point. Should only affect forked processes at worst, but if the tests are failing (because `example.run` raises an error) then we wouldn't be using the result of that test run anyways.